sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,333 @@ const createDomNode = (nodeName, attributes) => {
     return node;
 };
 
+/**
+ * Generate and return html structure
+ * for a page section result.
+ *
+ * @param {Object} sectionData object containing the result data
+ * @param {String} page_link link of the main page. It is used to construct the section link
+ */
+const get_section_html = (sectionData, page_link) => {
+    let section_template =
+        '<a href="<%= section_link %>"> \
+            <div class="outer_div_page_results" id="<%= section_id %>"> \
+                <span class="search__result__subheading"> \
+                    <%= section_subheading %> \
+                </span> \
+                <p class="search__result__content"> \
+                    <%= section_content %> \
+                </p> \
+            </div> \
+        </a> \
+        <br class="br-for-hits">';
+
+    let section_subheading = sectionData._source.title;
+
+    if (
+        sectionData.highlight["sections.title"] !== undefined &&
+        sectionData.highlight["sections.title"].length >= 1
+    ) {
+        section_subheading = sectionData.highlight["sections.title"][0];
+    }
+
+    let section_content =
+        sectionData._source.content.substring(0, 100) + " ...";
+
+    if (
+        sectionData.highlight["sections.content"] !== undefined &&
+        sectionData.highlight["sections.content"].length >= 1
+    ) {
+        section_content =
+            "... " + sectionData.highlight["sections.content"][0] + " ...";
```

Comment:
Is there a reason we're accessing only the first item of the list here? Should this be a constant about how many to include?

Reply:
We were getting only one result per section. So there has to be only one element in the array.
But I have refactored the code now and it is much better. We can show as many results from a section as we want. Here's a demo:
![Peek 2019-07-09 13-42](https://user-images.githubusercontent.com/29149191/60871220-f0f1b800-a24f-11e9-8ba8-32f72044e92a.gif)
I have used it with maximum 3 results from a section

Reply:
Currently I have used the approach of getting the value at index 0 directly everywhere.
I will be refactoring them also. (search.html and main site search)